src/migrations/squash.rs

```diff
@@ -17,29 +38,92 @@ pub async fn main(cli: &mut Connection, options: &Options,
         create.non_interactive,
     ).await?;
     let migrations = migration::read_all(&ctx, true).await?;
-    let Some(db_rev) = up_to_date_check(cli, &ctx, &migrations).await? else {
+    let Some(db_rev) = migrations_applied(cli, &ctx, &migrations).await? else {
         return Err(ExitCode::new(3).into());
     };
+    let needs_fixup = needs_fixup(cli, &ctx).await?;
+
     if db_rev == "initial" {
         echo!("No migrations exist. Nothing to do.");
         return Ok(());
     }
-    if migrations.len() == 1 {
+    if migrations.len() == 1 && !needs_fixup {
         echo!("Only single revision exists. Nothing to do.");
         return Ok(());
     }
     if !create.non_interactive {
-        confirm_squashing(&db_rev).await?;
+        cli.ping_while(confirm_squashing(&db_rev)).await?;
+    }
+
+    let squashed = create_revision(cli, &ctx, create).await?;
+
+    let key = MigrationKey::Fixup {
+        target_revision: squashed.id()?.to_owned(),
+    };
+    let fixup = if needs_fixup {
+        if create.non_interactive || cli.ping_while(want_fixup()).await? {
+            let parent = Some(&db_rev[..]);
+            Some(normal_migration(cli, &ctx, key, parent, create).await?)
+        } else {
+            None
+        }
```

Comment:
If the user refused the fixup to include latest schema changes, we should at least create an empty fixup for the squash itself?